pkg/gui/commits_panel.go

```diff
@@ -8,172 +8,395 @@ import (
+
 	if len(gui.State.Commits) == 1 {
-		return gui.createErrorPanel(g, gui.Tr.SLocalize("YouNoCommitsToSquash"))
+
+		err := gui.createErrorPanel(gui.Tr.SLocalize("YouNoCommitsToSquash"))
+		if err != nil {
+			gui.Log.Errorf("Failed to create error panel at handleCommitFixup: %s\n", err)
+			return err
+		}
+
+		return nil
 	}
+
 	if gui.anyUnStagedChanges(gui.State.Files) {
-		return gui.createErrorPanel(g, gui.Tr.SLocalize("CantFixupWhileUnstagedChanges"))
+		err := gui.createErrorPanel(gui.Tr.SLocalize("CantFixupWhileUnstagedChanges"))
+		if err != nil {
+			gui.Log.Errorf("Failed to create error panel at handleCommitFixup: %s\n", err)
+			return err
+		}
+
+		return nil
 	}
+
 	branch := gui.State.Branches[0]
-	commit, err := gui.getSelectedCommit(g)
+
+	commit, err := gui.getSelectedCommit()
 	if err != nil {
+		gui.Log.Errorf("Failed to get selected commit: %s\n", err)
 		return err
 	}
+
 	message := gui.Tr.SLocalize("SureFixupThisCommit")
-	gui.createConfirmationPanel(g, v, gui.Tr.SLocalize("Fixup"), message, func(g *gocui.Gui, v *gocui.View) error {
-		if err := gui.GitCommand.SquashFixupCommit(branch.Name, commit.Sha); err != nil {
-			return gui.createErrorPanel(g, err.Error())
-		}
-		if err := gui.refreshCommits(g); err != nil {
-			panic(err)
-		}
-		return gui.refreshStatus(g)
-	}, nil)
+
+	err = gui.createConfirmationPanel(v, gui.Tr.SLocalize("Fixup"), message,
+		func(g *gocui.Gui, v *gocui.View) error {
+
+			err := gui.GitCommand.SquashFixupCommit(branch.Name, commit.Sha)
+			if err != nil {
+
+				err = gui.createErrorPanel(err.Error())
+				if err != nil {
+					gui.Log.Errorf("Failed to create error panel at handleCommitFixup: %s\n", err)
+					return err
+				}
+
+				return nil
+			}
+
+			err = gui.refreshCommits()
+			if err != nil {
+				gui.Log.Errorf("Failed to refresh commits at handleCommitFixup: %s\n", err)
+				return err
+			}
+
+			err = gui.refreshStatus()
+			if err != nil {
+				gui.Log.Errorf("Failed to refresh status at handleCommitFixup: %s\n", err)
+				return err
+			}
+
+			return nil
+		}, nil)
+	if err != nil {
+		gui.Log.Errorf("Failed to create confirmation panel at handleCommitFixup: %s\n", err)
+		return err
+	}
+
 	return nil
 }
 
+// handleRenameCommit is called when a user wants to rename a commit.
+// g and v are passed by the gocui library but only v is used.
+// If anything goes wrong it returns an error.
 func (gui *Gui) handleRenameCommit(g *gocui.Gui, v *gocui.View) error {
-	if gui.getItemPosition(gui.getCommitsView(g)) != 0 {
-		return gui.createErrorPanel(g, gui.Tr.SLocalize("OnlyRenameTopCommit"))
-	}
-	return gui.createPromptPanel(g, v, gui.Tr.SLocalize("renameCommit"), func(g *gocui.Gui, v *gocui.View) error {
-		if err := gui.GitCommand.RenameCommit(v.Buffer()); err != nil {
-			return gui.createErrorPanel(g, err.Error())
-		}
-		if err := gui.refreshCommits(g); err != nil {
-			panic(err)
+
+	if gui.getItemPosition(v) != 0 {
```

Comment:
Were @dawidd6's concerns about passing v here valid? If so we'll need to swap this out for `gui.getCommitsView(g)`

Reply:
It was, but I solved it at the root of the problem, see handleMenuPress in menu_panel.go
```
vv, err := gui.g.View(gui.State.Keys[lineNumber].ViewName)
if err != nil {
	gui.Log.Errorf("Failed to get view at handleMenuPress: %s\n", err)
	return err
}

err = gui.State.Keys[lineNumber].Handler(gui.g, vv)
if err != nil {
	gui.Log.Errorf("Failed to call handler for %s\n at handleMenuPress: %s\n", gui.State.Keys[lineNumber].Description, err)
	return err
}
```

Reply:
If it's really working as intended then good work.